app/controllers/application_controller.rb

```diff
@@ -4,8 +4,18 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :set_app_info_headers
 
+  private
+
   def set_app_info_headers
     headers["X-GitHub-Repository"] = "https://github.com/department-of-veterans-affairs/vets-api"
     headers["X-Git-SHA"] = AppInfo::GIT_REVISION
   end
+
+  def require_login
+    unless session[:user]
+      flash[:after_login_controller] = request.parameters["controller"]
```

Comment:
Perhaps, rather than a redirect, we should raise an exception, and then have that exception caught in application controller as a particular type of exception, with JSON response body. APIs and redirects don't work well.